orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py

```diff
@@ -660,3 +660,53 @@ def get_inner_module_call_result(x, device, use_ort):
     result_ort = get_inner_module_call_result(x.detach(), 'cpu', True)
     result_pth = get_inner_module_call_result(x.detach(), 'cpu', False)
     compare_tensor_list(result_ort, result_pth)
+
+
+def test_Share_Input():
+    class TwoOutputFunction(torch.autograd.Function):
+        @staticmethod
+        # bias is an optional argument
+        def forward(ctx, x, y):
+            ctx.save_for_backward(x, y)
+            w = x + y
+            z = x * y
+            return w, z
+
+        @staticmethod
+        def backward(ctx, dw, dz):
+            x, y = ctx.saved_tensors
+            dx = dw * 1.0 + dz * y
+            dy = dw * 1.0 + dz * x
+            return dx, dy
+
+    class TwoOutputModel(torch.nn.Module):
+        def __init__(self, output_size):
+            super(TwoOutputModel, self).__init__()
+            self.fun = TwoOutputFunction.apply
+            self.bias = Parameter(torch.empty(
+                output_size,
+                device=torch.cuda.current_device(),
+                dtype=torch.float))
+
+            # Always initialize bias to zero.
+            with torch.no_grad():
+                self.bias.uniform_()
+
+        def forward(self, x):
+            a, b = self.fun(x, self.bias)
```

Comment:
As we synced offline, we'd better try a case x.requires_grad=True.

Reply:
Done.